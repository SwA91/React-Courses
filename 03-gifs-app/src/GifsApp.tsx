import { CustomHeader } from "./shared/components/CustomHeader";
import { SearchBar } from "./shared/components/SearchBar";
import { PreviousSearches } from "./gifs/components/PreviousSearches";
import { GifList } from "./gifs/components/GifList";
import { useState } from "react";
import { getGifsByQuery } from "./gifs/actions/get-gifs-by-query.action";
import type { Gif } from "./gifs/interfaces/gif.interface";

export const GifsApp = () => {
  const [gifs, setGifs] = useState<Gif[]>([]);
  const [previousTerms, setPreviousTerms] = useState<string[]>([]);
  const handleTermClicked = (term: string) => {
    console.log({ term });
  };

  const handleSearch = async (query: string) => {
    query = query.trim().toLowerCase();
    if (query.length === 0) return;
    if (previousTerms.includes(query)) return;
    setPreviousTerms([query, ...previousTerms].slice(0, 6));

    const gifs = await getGifsByQuery(query);
    setGifs(gifs);
  };
  return (
    <>
      {/* header */}
      <CustomHeader
        title="buscador de gifs"
        description="descubre y comparte gifs"
      ></CustomHeader>

      {/* search */}
      <SearchBar onQuery={handleSearch} placeholder="Busca tu GIF"></SearchBar>

      {/* busquedas previas */}
      <PreviousSearches
        onLabelClicked={handleTermClicked}
        searches={previousTerms}
      ></PreviousSearches>

      {/* Gifs */}
      <GifList gifs={gifs}></GifList>
    </>
  );
};
