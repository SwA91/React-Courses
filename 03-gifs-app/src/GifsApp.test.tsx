import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { GifsApp } from "./GifsApp";

describe("GifsApp", () => {
  test("should render componente propertly", () => {
    const { container } = render(<GifsApp></GifsApp>);
    expect(container).toMatchSnapshot();
    // provocamos un fallo para probar nuestro build
    // expect(2).toBe(1);
  });
});
