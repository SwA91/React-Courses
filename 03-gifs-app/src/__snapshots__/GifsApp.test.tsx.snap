// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsApp > should render componente propertly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      buscador de gifs
    </h1>
    <p>
      descubre y comparte gifs
    </p>
  </div>
  <div
    class="search-container"
  >
    <input
      placeholder="Busca tu GIF"
      type="text"
      value=""
    />
    <button>
      Buscar
    </button>
  </div>
  <div
    class="previous-searches"
  >
    <h2>
      busquedas previas
    </h2>
    <ul
      class="previous-searches-list"
    />
  </div>
  <div
    class="gifs-container"
  />
</div>
`;
